src/components/Devices/Devices.tsx

```diff
@@ -71,12 +70,18 @@ function Devices() {
     if (controller.getUserOidcToken()) {
       getCollection();
     } else {
-      navigate({
-        pathname: '/login',
-        search:   searchParams.toString(),
+      (new Promise((resolve) => { setTimeout(resolve, 5000); })).then(controller.getUserOidcToken).then((token) => {
```

Comment:
I assume that this part of code is delaying the checking token by 5 seconds. (I think this may be too long?) 

Also maybe we can rewrite this part as recursive check, eg every 1 second we check if token is ready for 3 times and if within 3 tries if it exist, we stop next recursive call and execute `getCollection` or if it failed to check after 3 tries, we execute redirect logic? 

But for the time being, maybe we go with what we have

Reply:
We can use interval checking or exponential backoff type behavior here. Feel free to open an issue regarding this